test.py

```diff
@@ -1704,19 +1717,22 @@ async def main() -> int:
     options = parse_cmd_line()
 
     open_log(options.tmpdir, f"test.py.{'-'.join(options.modes)}.log", options.log_level)
-
+    setup_cgroup(options.gather_metrics)
     await find_tests(options)
     if options.list_tests:
         print('\n'.join([f"{t.suite.mode:<8} {type(t.suite).__name__[:-9]:<11} {t.name}"
                          for t in TestSuite.all_tests()]))
         return 0
 
     signaled = asyncio.Event()
+    stop_event = asyncio.Event()
+    run_resource_watcher(options.gather_metrics, signaled, stop_event)
 
     setup_signal_handlers(asyncio.get_running_loop(), signaled)
 
     try:
         await run_all_tests(signaled, options)
+        stop_event.set()
```

Comment:
you need to wait for monitor_resources to finish as well as it has 2 seconds of sleep in the loop. also if it crashes you will be able to get stack trace only at the join